src/euler_types.jl

```diff
@@ -8,25 +8,52 @@
 #########################
 # Single axis rotations #
 #########################
+
+for axis in [:X, :Y, :Z]
+    RotType = Symbol("Rot" * string(axis))
+    @eval begin
+        struct $RotType{T} <: Rotation{3,T}
+            theta::T
+            $RotType{T}(theta) where {T} = new{T}(theta)
+            $RotType{T}(r::$RotType) where {T} = new{T}(r.theta)
+        end
+
+        @inline $RotType(theta::T) where {T} = $RotType{T}(theta)
+        @inline $RotType(r::$RotType{T}) where {T} = $RotType{T}(r)
+
+        @inline convert(::Type{R}, r::$RotType) where {R<:$RotType} = R(r)
+        @inline convert(::Type{R}, r::R) where {R<:$RotType} = r
+
+        @inline (::Type{R}){R<:$RotType}(t::NTuple{9}) = error("Cannot construct a cardinal axis rotation from a matrix")
+
+        @inline Base.:*(r1::$RotType, r2::$RotType) = $RotType(r1.theta + r2.theta)
+
+        @inline inv(r::$RotType) = $RotType(-r.theta)
+
+        # define null rotations for convenience
+        @inline eye(::Type{$RotType}) = $RotType(0.0)
+        @inline eye{T}(::Type{$RotType{T}}) = $RotType{T}(zero(T))
+    end
+end
+
+function Base.rand{R <: Union{RotX,RotY,RotZ}}(::Type{R})
```

Comment:
Either way for me. I don't currently have a good mental model for costs related to method tables. @andyferris, any thoughts?